video/out/wayland_common.c

```diff
@@ -1207,7 +1249,12 @@ static void registry_handle_add(void *data, struct wl_registry *reg, uint32_t id
     struct vo_wayland_state *wl = data;
 
     if (!strcmp(interface, wl_compositor_interface.name) && (ver >= 4) && found++) {
-        wl->compositor = wl_registry_bind(reg, id, &wl_compositor_interface, 4);
+#ifdef HAVE_WAYLAND_1_22
+        ver = MPMIN(ver, 6); /* Cap at 6 in case new events are added later. */
+#else
+        ver = 4;
+#endif
+        wl->compositor = wl_registry_bind(reg, id, &wl_compositor_interface, ver);
```

Comment:
I don't think this is needed? Or would a compositor that's compiled against an old version of libwayland still deliver version 6?

Reply:
Maybe mpv was compiled with an old libwayland then libwayland has been upgraded. Maybe libwayland is linked statically. Maybe mpv is in a sandbox and using a different libwayland library. etc etc

Reply:
Fair enough.